### PR TITLE
Untibble

### LINKED ISSUE #609: Move tibble to "Suggests:"
Excluding `igraph` and "`Suggests:`", `tibble` now stands out as `drake`'s heaviest strict dependency. I think we should make `tibble` optional.

Internally, we could replace `tibble()` and `as_tibble()` with functions that return `data.frame`s if `tibble` is not installed. We would need to do this in `R/*.R` and `tests/testthat/*.R`.

@bpbond, are you interested in taking on the implementation?

### PR DESCRIPTION
**Pull request by bpbond**
Closes #609 

# Summary

Per discussion in #609 create `weak_tibble` and `weak_as_tibble` functions that will use `tibble::tibble` if it's available but if not fall back to `data.frame`.

# Related GitHub issues and pull requests

- Ref: #609 

# Checklist

- [X] I have read `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CONDUCT.md), and I agree to follow its rules.
- [x] I have listed any substantial changes in the [development news](https://github.com/ropensci/drake/blob/master/NEWS.md).
- [X] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/ropensci/drake/tree/master/tests/testthat) to confirm that any new features or functionality work correctly.
- [X] I have tested this pull request locally with `devtools::check()`
- [X] This pull request is ready for review.
- [ ] I think this pull request is ready to merge.

### CODE REVIEW

**Comment by bpbond** [comment]
@wlandau I have done a first pass at this but the existing test code has some funky (for me) things involving quasi quotation etc. that cause my implementation of `weak_tibble` to die:
```
test-cache.R:7: error: clean() removes the correct files
object 'targets' not found
1: drake_plan(a = file_in("a.txt"), b = knitr_in("b.txt"), d = writeLines("123", 
       file_out("d.rds")), strings_in_dots = "literals") at /Users/d3x290/Documents/Work/Code/drake/tests/testthat/test-cache.R:7
2: weak_tibble(target = targets, command = commands) at /Users/d3x290/Documents/Work/Code/drake/R/drake_plan.R:174
3: do.call(tibble::tibble, args) at /Users/d3x290/Documents/Work/Code/drake/R/utils.R:297
4: (function (...) 
   {
       xs <- quos(..., .named = TRUE)
       as_tibble(lst_quos(xs, expand = TRUE))
   })(target = targets, command = commands)
5: as_tibble(lst_quos(xs, expand = TRUE))
6: lst_quos(xs, expand = TRUE)
7: eval_tidy(xs[[i]], unique_output)
```
I'll come back to this and experiment more, but this is not an area I'm super comfortable in and welcome any feedback on this:
```R
# weak_tibble - use tibble() if available but fall back to 
# data.frame() if necessary. Note that testing code may supply
# a `.force_df` argument that must be removed
weak_tibble <- function(...) {
  args <- eval(substitute(alist(...)))
  .force_df <- eval.parent(args$.force_df, 1)
  if (is.null(.force_df)) {
    .force_df <- FALSE
  } else {
    args$.force_df <- NULL
  }

  if (!.force_df & require(tibble, quietly = TRUE)) {
    do.call(tibble::tibble, args)
  } else {
    do.call(data.frame, c(args, stringsAsFactors = FALSE))
  }
}
```
Thanks!

**Comment by wlandau** [comment]
I think we make `.force_df` a formal argument and leave `...` as is.

``` r
weak_tibble <- function(..., .force_df = FALSE) {
  no_tibble <- !suppressWarnings(require("tibble", quietly = TRUE))
  if (.force_df || no_tibble) {
    data.frame(..., stringsAsFactors = FALSE)
  } else {
    tibble::tibble(...)
  }
}

weak_tibble()
#> # A tibble: 0 x 0

weak_tibble(a = 1:2, b = 3:4, c = c("five", "six"))
#> # A tibble: 2 x 3
#>       a     b c    
#>   <int> <int> <chr>
#> 1     1     3 five 
#> 2     2     4 six

weak_tibble(.force_df = TRUE)
#> data frame with 0 columns and 0 rows

weak_tibble(a = 1:2, b = 3:4, c = c("five", "six"), .force_df = TRUE)
#>   a b    c
#> 1 1 3 five
#> 2 2 4  six
```

<sup>Created on 2018-12-15 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by bpbond** [comment]
Hmm, that's actually what I started with, then changed when ran into testing trouble. OK, will try again. Thanks.

**Comment by bpbond** [comment]
Don't know what happened in the original testing of that approach--I must have made a mistake somewhere and mis-interpreted what was happening. Argh. Pushing.

File: tests/testthat/test-utils.R [hunk 1]
```diff
@@ -51,3 +51,43 @@ test_with_dir("operators", {
   expect_true(is.numeric(Inf %||NA% "b"))
   expect_false(is.na(NA %||NA% "b"))
 })
+
+test_with_dir("weak_tibble", {
```
Comment (wlandau) [inline tests/testthat/test-utils.R]: I do not think we need to focus on class names in this test, and I think we can skip the `tibble` part if `tibble` is not installed. I picture something like this:

```r
test_with_dir("weak tibbles without tibble", {
  skip_on_cran()
  out <- weak_tibble(.force_df = TRUE)
  expect_equal(out, data.frame())
  expect_equal(weak_as_tibble(list()), data.frame())
  out <- weak_tibble(a = 1:2, b = c("x", "y"), .force_df = TRUE)
  exp <- data.frame(a = 1:2, b = c("x", "y"), stringsAsFactors = FALSE)
  expect_equal(out, exp)
  out <- weak_as_tibble(list(a = 1:2, b = c("x", "y")))
  expect_equal(out, exp)
})

test_with_dir("weak tibbles with tibble", {
  skip_on_cran()
  skip_if_not_installed("tibble")
  out <- weak_tibble()
  expect_equal(out, tibble::tibble())
  expect_equal(weak_as_tibble(list()), tibble::tibble())
  out <- weak_tibble(a = 1:2, b = c("x", "y"))
  exp <- tibble::tibble(a = 1:2, b = c("x", "y"))
  expect_equal(out, exp)
  out <- weak_as_tibble(list(a = 1:2, b = c("x", "y")))
  expect_equal(out, exp)
})
```

**Comment by wlandau** [comment]
Thanks again, @bpbond!

File: tests/testthat/test-utils.R [hunk 2]
```diff
@@ -51,3 +51,20 @@ test_with_dir("operators", {
...
```
Comment (wlandau) [inline tests/testthat/test-utils.R]: I really like the way you compacted this test.

**Pull request merged**